ddtrace/llmobs/_evaluators/runner.py

```diff
@@ -92,25 +89,33 @@ def enqueue(self, span_event: Dict, span: Span) -> None:
                 return
             self._buffer.append((span_event, span))
 
-    def periodic(self) -> None:
+    def periodic(self, _wait_syncronously=False) -> None:
+        """
+        :param bool _wait_syncronously: if `True`, each evaluator is run for each span in the buffer
+        synchronously. This param is only set to `True` for when the evaluator runner is stopped by the LLM Obs
+        instance on process exit and we want to block until all spans are evaluated and metrics are submitted.
+        """
         with self._lock:
             if not self._buffer:
                 return
             span_events_and_spans = self._buffer  # type: list[tuple[Dict, Span]]
             self._buffer = []
 
         try:
-            self.run(span_events_and_spans)
+            if not _wait_syncronously:
+                for evaluator in self.evaluators:
+                    self.executor.map(
+                        lambda span_event: evaluator.run_and_submit_evaluation(span_event),
+                        [
+                            span_event
+                            for span_event, span in span_events_and_spans
+                            if self.sampler.sample(evaluator.LABEL, span)
+                        ],
+                    )
+            else:
+                for evaluator in self.evaluators:
+                    for span_event, span in span_events_and_spans:
+                        if self.sampler.sample(evaluator.LABEL, span):
+                            evaluator.run_and_submit_evaluation(span_event)
         except RuntimeError as e:
             logger.debug("failed to run evaluation: %s", e)
-
```

Comment:
This function is not re-used anywhere, just going to inline the code here in `periodic`